Userland/Applications/PixelPaint/main.cpp

```diff
@@ -53,8 +54,10 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     TRY(main_widget->initialize_menubar(*window));
 
     window->on_close_request = [&]() -> GUI::Window::CloseRequestDecision {
-        if (main_widget->request_close())
+        if (main_widget->request_close()) {
+            GUI::DynamicWidgetContainer::close_all_detached_windows();
```

Comment:
This should not be necessary. Please figure out how to call that function at application exit; or better yet, for each separate container widget when it's destructed (RAII).

Reply:
For the moment i was not able to find a reliable working solution e.g. closing all windows within the destructor. That destructor somehow never was called when the widget has been cunstructed via a gml file - I have no idea whats the reason behind that.
For the moment i close all the open windows in the destructor as when the widget was not constructed via gml and the old solution where they are closed via the static function.